File: app/src/main/java/me/imzack/app/end/injector/component/GuideComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.GuidePresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.GuideActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = GuidePresenterModule.class)
public interface GuideComponent {

    void inject(GuideActivity guideActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/GuideComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.GuidePresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.GuideActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(GuidePresenterModule::class))
interface GuideComponent {

    fun inject(guideActivity: GuideActivity)
}
